Cooper.DAO/MessageDAO/MessageDAO.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using Cooper.DAO.Mapping;
+using Cooper.DAO.Models;
+using Cooper.ORM;
+using Cooper.Services.Interfaces;
+using NLog;
+using System.Collections.Generic;
+using System.Linq;
+using System.Data.Common;
+using Oracle.ManagedDataAccess.Client;
+
+namespace Cooper.DAO
+{
+    public class MessageDAO : IMessageDAO
+    {
+        private readonly CRUD crud;
+        private readonly Logger logger;
+        private readonly DbConnect dbConnect;
+
+        private string table;
+        private string idColumn;
+        private HashSet<string> attributes;
+
+        public MessageDAO(IConfigProvider configProvider)
+        {
+            crud = new CRUD(configProvider);
+            dbConnect = new DbConnect(configProvider);
+            logger = LogManager.GetLogger("CooperLoger");
+
+            table = "MESSAGES";
+            idColumn = "ID";
+            attributes = new HashSet<string>()
+            {
+                "ID", "IDSENDER", "IDCHAT", "CONTENT", "CREATEDATE", "ISREAD"
+            };
+        }
+
+        public IList<MessageDb> GetAllMessagesByChatId(long chatId)
+        {
+            IList<MessageDb> messages = null;
+
+            var whereAttributes = new Dictionary<string, object>()
+            {
+                {"IDCHAT", chatId }
+            };
+
+            IList<EntityORM> entities = Get(table, attributes, whereAttributes);
+
+            if (entities != null)
+            {
+                messages = new List<MessageDb>();
+                foreach (var entity in entities)
+                {
+                    EntityMapping.Map(entity, out MessageDb message);
+                    messages.Add(message);
+                }
+            }
+
+            return messages;
+        }
+
+        public bool ReadNewMessages(IList<long> messages)
+        {
+            string sqlExpression = $"UPDATE {table} SET ISREAD=\'y\' ";
```

Comment:
Can't DBTools generete this query with "IN"?

Reply:
No. Logic for generating update query for more than one entities is abcent as well as logic in CRUD for updating several entities